ivy.el

```diff
@@ -253,6 +253,12 @@ It is a list of (CALLER . HEIGHT).  CALLER is a caller of
 If `(minibuffer-depth)' equals this, `ivy-completing-read' will
 act as if `ivy-completing-read-handlers-alist' is empty.")
 
+(defvar counsel--search-commands '(counsel-git-grep
```

Comment:
Is there any reason why this variable is defined so far away from its call site? What is this variable's purpose? What other use-cases for this variable are there? Wouldn't it be safer to initialise this variable to `nil` and have something like
```el
(cl-pushnew #'counsel-git-grep counsel--search-commands)
```
after every search command is defined?

Reply:
1) That's my fault. 
2) make more customizable, without overriding whole function such as ivy--highlight-default
3) i am not looking so far, but store commands in variable more correctly. For example after appending new command (like spacemacs did), we don't need edit source code or override func (how i said in previous comment)
4) i'm agree. it's better.